diff --git a/Util/fitsutil/fits_version_check.C b/Util/fitsutil/fits_version_check.C
index 8a8bd401e..7df44d500 100644
--- a/Util/fitsutil/fits_version_check.C
+++ b/Util/fitsutil/fits_version_check.C
@@ -18,7 +18,15 @@ void fits_version_check (bool verbose)
     float version = 0.0;
     fits_get_version (&version);
 
-    float lib_version = CFITSIO_MAJOR + CFITSIO_MINOR/pow(10,ceil(log10(CFITSIO_MINOR)));
+#if (CFITSIO_MAJOR < 4)
+    // Older-style "X.YY" versioning
+    float lib_version = CFITSIO_VERSION;
+#else
+    // Newer "X.Y.Z" versioning.  fits_get_version now returns
+    // X + 0.01*Y + 0.0001*Z according to:
+    // https://heasarc.gsfc.nasa.gov/docs/software/fitsio/c/c_user/node52.html#ffvers
+    float lib_version = CFITSIO_MAJOR + 0.01*CFITSIO_MINOR + 0.0001*CFITSIO_MICRO;
+#endif
 
     if (verbose)
         cerr << "fits_version_check:"
