diff --git a/More/python/Makefile.am b/More/python/Makefile.am
index c633962af..7b18681a8 100644
--- a/More/python/Makefile.am
+++ b/More/python/Makefile.am
@@ -19,7 +19,7 @@ _psrchive_la_LIBADD = $(top_builddir)/More/libpsrmore.la \
 LOCAL_CFLAGS = -I$(top_builddir)/local_include -I$(top_builddir) -I$(top_srcdir)/Util/meal/src/util
 
 SWIG_FILE = $(srcdir)/psrchive.i
-SWIG_FLAGS = -Wall -python $(LOCAL_CFLAGS)
+SWIG_FLAGS = -Wall -python -flatstaticmethod $(LOCAL_CFLAGS)
 SWIG_DEPS = $(DEPDIR)/swig.deps
 
 psrchive_wrap.cxx: $(SWIG_FILE) $(top_builddir)/config.h
--- More/python/Makefile.in	2026-08-13 08:24:04.586992694 +1200
+++ More/python/Makefile.in	2026-08-14 19:08:18.022459846 +1200
@@ -473,7 +473,7 @@
 
 LOCAL_CFLAGS = -I$(top_builddir)/local_include -I$(top_builddir) -I$(top_srcdir)/Util/meal/src/util
 SWIG_FILE = $(srcdir)/psrchive.i
-SWIG_FLAGS = -Wall -python $(LOCAL_CFLAGS)
+SWIG_FLAGS = -Wall -python -flatstaticmethod $(LOCAL_CFLAGS)
 SWIG_DEPS = $(DEPDIR)/swig.deps
 local_includedir = $(top_builddir)/local_include
 LOCAL_INCLUDE = $(addprefix $(local_includedir)/, $(HEADERS))
