--- a/More/python/Makefile.in	2023-02-13 18:27:51.000000000 -0700
+++ b/More/python/Makefile.in	2023-02-13 18:28:23.000000000 -0700
@@ -470,7 +470,7 @@
                       $(top_builddir)/Util/epsic/src/libepsic.la \
                       @PYTHON_LDFLAGS@ 
 
-SWIG_FLAGS = -Wall -python -I$(top_builddir)/local_include -I$(top_builddir)
+SWIG_FLAGS = -Wall -python -flatstaticmethod -I$(top_builddir)/local_include -I$(top_builddir)
 local_includedir = $(top_builddir)/local_include
 LOCAL_INCLUDE = $(addprefix $(local_includedir)/, $(HEADERS))
 
