diff --git a/More/Plotting/StokesCovariancePlot.C b/More/Plotting/StokesCovariancePlot.C
index 506bfce57..3f3a2a34a 100644
--- a/More/Plotting/StokesCovariancePlot.C
+++ b/More/Plotting/StokesCovariancePlot.C
@@ -279,9 +279,9 @@ void Pulsar::StokesCovariancePlot::get_profiles (const Archive* data) try
     
   }
 
-  set (plotter.plot_sci, plot_colours);
-  set (plotter.plot_sls, plot_lines);
-  set (plotter.plot_slw, plot_widths);
+  ::set (plotter.plot_sci, plot_colours);
+  ::set (plotter.plot_sls, plot_lines);
+  ::set (plotter.plot_slw, plot_widths);
 }
 
  catch (Error& error)
diff --git a/Util/units/ReferenceAble.C b/Util/units/ReferenceAble.C
index 1820e0691..1bb32fcb1 100644
--- a/Util/units/ReferenceAble.C
+++ b/Util/units/ReferenceAble.C
@@ -53,7 +53,7 @@ static pthread_mutex_t* get_mutex ()
 {
   static pthread_mutex_t* mutex = 0;
   if (!mutex)
-    mutex = recursive_mutex ();
+    mutex = ::recursive_mutex ();
   return mutex;
 }
 
