diff --git a/Util/genutil/CommandLine.C b/Util/genutil/CommandLine.C
index 52db45bb3..fcd5addca 100644
--- a/Util/genutil/CommandLine.C
+++ b/Util/genutil/CommandLine.C
@@ -100,7 +100,7 @@ void find_and_erase (C& container, const E& element)
     container.erase (found);
 }
 
-void CommandLine::Menu::remove (Item* i)
+void CommandLine::Menu::remove (const Item* i)
 {
   find_and_erase (item, i);
 }
diff --git a/Util/genutil/CommandLine.h b/Util/genutil/CommandLine.h
index c88a921f7..18d69c526 100644
--- a/Util/genutil/CommandLine.h
+++ b/Util/genutil/CommandLine.h
@@ -352,7 +352,7 @@ namespace CommandLine {
     virtual void add (Item*);
 
     //! Remove an item from the menu
-    virtual void remove (Item*);
+    virtual void remove (const Item*);
 
     //! Add a Value with only a single letter name
     template<typename T>
