--- psrchive.i	2026-02-26 21:43:35.000000000 +1300
+++ psrchive.i	2026-08-16 08:00:18.620399233 +1200
@@ -5,12 +5,33 @@
 
 %{
 #define SWIG_FILE_WITH_INIT
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include "numpy/arrayobject.h"
 
 #ifndef PSRCHIVE_PyArray_DATA
   #define PSRCHIVE_PyArray_DATA(a) PyArray_DATA(a)
 #endif
 
+#include <Python.h>
+
+/*
+  Python 2 and 3 compatibility macros
+  String handling changed between python 2 and 3.
+*/
+#if PY_MAJOR_VERSION >= 3
+  #define compat_PyString_FromString PyUnicode_FromString
+  #define compat_PyString_AsString   PyUnicode_AsUTF8
+  #define compat_PyString_Check      PyUnicode_Check
+#else
+  #define compat_PyString_FromString PyString_FromString
+  #define compat_PyString_AsString   PyString_AsString
+  #define compat_PyString_Check      PyString_Check
+#endif
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include "Reference.h"
 #include "Pulsar/IntegrationManager.h"
 #include "Pulsar/IntegrationExpert.h"
@@ -222,17 +243,11 @@
 // Convert various enums to/from string
 %define %map_enum(TYPE)
 %typemap(out) Signal:: ## TYPE {
-    $result = PyString_FromString( TYPE ## 2string($1).c_str());
+    $result = compat_PyString_FromString( TYPE ## 2string($1).c_str());
 }
 %typemap(in) Signal:: ## TYPE %{
     try {
-        // String handling changed between python 2 and 3.
-        // This approach should be OK for 2.7 and 3.3+
-#if PY_VERSION_HEX >= 0x03030000
-        const char *typestr = PyUnicode_AsUTF8($input);
-#else
-        const char *typestr = PyString_AsString($input);
-#endif
+        const char *typestr = compat_PyString_AsString($input);
         if (typestr!=NULL) {
             $1 = Signal::string2 ## TYPE (typestr);
         }
@@ -733,7 +748,7 @@
         }
         else
         {
-            PyObject *result = (PyObject *)PyString_FromString("undefined");
+            PyObject *result = (PyObject *)compat_PyString_FromString("undefined");
             return (PyObject *)result;
         } 
     }
